.github/workflows/scripts/check-collector-module-version.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+#   Copyright The OpenTelemetry Authors
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#       http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+#
+# verifies if the collector components are using the main core collector version
+# as a dependency.
+#
+set -eu -o pipefail
+
+# Return the collector main core version
+get_collector_version() {
+   collector_module="$1"
+   main_mod_file="$2"
+
+   if grep -q "$collector_module" "$main_mod_file"; then
+      grep "$collector_module" "$main_mod_file" | (read mod version;
+         echo $version)
+   else
+      echo "Error: failed to retrieve the \"$collector_module\" version from \"$main_mod_file\"."
+      exit 1
+   fi
+}
+
+# Compare the collecor main core version against all the collector component
+# modules to verify that they are using this version as its dependency
+check_collector_version_correct() {
```

Comment:
This is checking all go.mod files against the main go.mod's, right? Perhaps using "versions" (plural) would be more appropriate for the function name?

```suggestion
check_collector_versions_correct() {
```

Reply:
That makes sense. Updated.